doc/reference/configuration/configuration_reference.rst

```diff
@@ -4767,6 +4768,37 @@ To learn more about the WAL configuration, check the :ref:`Persistence <configur
     | Default: 16777216
     | Environment variable: TT_WAL_QUEUE_MAX_SIZE
 
+..  _configuration_reference_wal_retention_period:
+
+..  confval:: wal.retention_period
+
+    **Since:** :doc:`3.1.0 </release/3.1.0>` (`Enterprise Edition <https://www.tarantool.io/compare/>`_ only)
+
+    The delay in seconds used to prevent the :ref:`Tarantool garbage collector <configuration_persistence_checkpoint_daemon>` from removing a :ref:`write-ahead log <internals-wal>` file after it has been closed.
+    If a node is restarted, ``wal.retention_period`` counts down from the last modification time of the write-ahead log file.
```

Comment:
I'd rather say, that `wal.retention_period counts down from the last modification time of the write-ahead log file, which means that node restart doesn't nullify the timeout`. Up to you